openfisca_core/base_functions.py

```diff
@@ -101,11 +101,12 @@ def last_duration_last_value(formula, simulation, period, *extra_params):
         for last_period, last_result in sorted(holder._array_by_period.iteritems(), reverse = True):
             if last_period.start <= period.start and (formula.function is None or last_period.stop >= period.stop):
                 output_period = periods.Period((last_period[0], period.start, last_period[2]))
+                assert output_period == period
                 if type(last_result) == np.ndarray and not extra_params:
-                    return output_period, last_result
+                    return last_result
                 elif last_result.get(extra_params):
-                    return output_period, last_result.get(extra_params)
+                    return last_result.get(extra_params)
     if formula.function is not None:
         return formula.exec_function(simulation, period, *extra_params)
     array = holder.default_array()
-    return period, array
+    return array
```

Comment:
I think this base function is not compatible with our new rule that says that a formula always give an answer for the period it has been asked for.

If it is confirmed that this base function is not used, we should remove it in this PR.